src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -266,6 +266,16 @@ Json::Value SotaUptaneClient::AssembleManifest() {
 
 bool SotaUptaneClient::hasPendingUpdates() const { return storage->hasPendingInstall(); }
 
+void SotaUptaneClient::initializePrimaryEcu() {
+  EcuSerials serials;
+  if (!storage->loadEcuSerials(&serials) || serials.size() == 0) {
+    throw std::runtime_error("Unable to load ECU serials after device registration.");
+  }
+
+  primary_ecu_serial_ = serials[0].first;
+  hw_ids.insert(serials[0]);
```

Comment:
As an alternative, `primary_ecu_serial_` and `hw_ids` could be initialized in SotaUptaneClient ctor if the additional params EcuSerial and HardwareIdentifier are added to it that equal to Uptane::EcuSerial::Unknown() and HardwareIdentifier::Unknown() respectively. In this case we won't this additional method and aktualizr-lite won't need to do additional post-initialization after SotaUptaneClient object is constructed.

Actually, I think we should get rid of  SotaUptaneClient::initialize() in aktualizr too.

Reply:
I took a fairly conservative approach to doing this and left the current initialize function in tact.

I think things are cleaner now. Thing are in 3 commit so you can see your comments being addressed. However, I think you'll want to squash them into 1 if you decide this is ready to be merged.

Reply:
thx, it looks great now.